src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -973,23 +978,17 @@ trait Scanners extends ScannersCommon {
     /** Convert current strVal, base to long value.
      *  This is tricky because of max negative value.
      *
-     *  Conversions in base 10 and 16 are supported. As a permanent migration
-     *  path, attempts to write base 8 literals except `0` emit a verbose error.
+     *  Conversions in base 2, 10 and 16 are supported.
      */
     def intVal(negated: Boolean): Long = {
       def intConvert: Long = {
         def malformed: Long = { syntaxError("malformed integer number") ; 0 }
         def tooBig: Long = { syntaxError("integer number too large") ; 0 }
-        val len = strVal.length
-        if (len == 0) {
-          if (base != 8) syntaxError("missing integer number")  // e.g., 0x;
-          0                                                     // 0 still looks like octal prefix
-        } else {
-          if (base == 8) {
-            if (settings.warnOctalLiteral)
-              deprecationWarning("Decimal integer literals should not have a leading zero. (Octal syntax is obsolete.)" , since="2.10")
-            base = 10
-          }
+        strVal.length match {
+        case 0 => syntaxError("missing integer number") ; 0L  // e.g., 0x;
```

Comment:
it seems from the tests that it is no longer possible to hit this condition;
`if (base != 10 && !isNumberSeparator(ch) && digit2int(ch, base) < 0)` on line 504 appears to always supersede it

Reply:
I would like to believe that. Previously, consuming the zero without storing it would result in the empty buffer. Thanks.